### DISCUSSION TITLE
INSTALL clarifications

### DISCUSSION THREAD

**Discussion by mariusa** [piku/piku#278] [Ideas]
Hi,

Per my understanding, one has to go through either ONE of
https://github.com/piku/piku/blob/master/docs/INSTALL.md 
or
https://github.com/piku/piku/blob/master/docs/ubuntu-18.04-bionic.md
or
https://github.com/piku/piku-bootstrap

and not do some steps in https://github.com/piku/piku/blob/master/docs/INSTALL.md  and then run https://github.com/piku/piku-bootstrap , right?

If so, I'd like to make a PR to 
1. move most of https://github.com/piku/piku/blob/master/docs/INSTALL.md 
into INSTALL-generic.md
2. rename ubuntu-18.04-bionic.md to INSTALL-ubuntu-18.04-bionic.md (and same for rapsbian)
3. add  INSTALL-ubuntu-20.md which just links to https://github.com/piku/piku-bootstrap

and later contribute an INSTALL-centos-stream-8.md

What do you think?

**Comment by chr15m** [comment]
Thanks for looking into this @mariusa. The install instructions are for those who want to install manually. The bootstrap is for people who want it done for them. So there are actually 3 ways to install piku on a server:

1. Follow the `INSTALL.md` manually for which ever platform.
2. Using cloud-init from @rcarmo to do it automatically at VPS build time: https://github.com/piku/cloud-init
3. Using piku-bootstrap to do it once your box is already provisioned: https://github.com/piku/piku-bootstrap

I am all for clarifying the docs. Maybe we need to make these various options clear?

**Reply by mariusa** [reply]
Thanks!

I've documented this and consolidated INSTALLs with https://github.com/piku/piku/pull/282
Please review, edit if necessary & merge.

After this is merged, I'll contribute an INSTALL-CentOS-9.md

**Comment by chr15m** [comment]
Thanks for doing this work. Let's wait for @rcarmo's feedback on these documentation changes.

**Comment by rcarmo** [comment]
It looks OK to me in general. There is one thing, though: I haven’t used the `uwsgi-piku` replacement service in a long while, I just [add a new `.ini` file to the system service](https://github.com/piku/cloud-init/blob/master/azure/cloud-init.yml#L79).

 I’m pretty sure that would work on all distros, and don’t actually remember why we started using a replacement service and disabling the system `uwsgi`…

**Comment by chr15m** [comment]
Maybe it was an older version of Ubuntu with an installed uwsgi that lacked a feature or didn't work the way Piku needed. I don't remember either. Maybe it's time to switch all the docs (and piku-bootstrap) back to using your ini file method.

**Comment by mariusa** [comment]
Please merge the current PR and we can have another for `uwsgi`. We can always make improvements, especially if you create a release now which will always work on old distros.

**Comment by mariusa** [comment]
FYI In Tyrant mode (set by default in /etc/uwsgi.ini) the Emperor will run the vassal using the UID/GID of the vassal configuration file

So I had to do this
```
chown piku:piku /etc/uwsgi.d/piku.ini 
```

The uid/gid lines in uwsgi piku.ini are ignored in Tyrant mode
